mirdata/groove_midi.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+"""Groove MIDI Loader
+
+The Groove MIDI Dataset (GMD) is composed of 13.6 hours of aligned MIDI and
+synthesized audio of human-performed, tempo-aligned expressive drumming.
+The dataset contains 1,150 MIDI files and over 22,000 measures of drumming.
+
+To enable a wide range of experiments and encourage comparisons between methods
+on the same data, Gillick et al. created a new dataset of drum performances
+recorded in MIDI format. They hired professional drummers and asked them to
+perform in multiple styles to a click track on a Roland TD-11 electronic drum kit.
+They also recorded the aligned, high-quality synthesized audio from the TD-11 and
+include it in the release.
+
+The Groove MIDI Dataset (GMD), has several attributes that distinguish it from
+existing ones:
+
+* The dataset contains about 13.6 hours, 1,150 MIDI files, and over 22,000
+measures of drumming.
+* Each performance was played along with a metronome set at a specific tempo
+by the drummer.
+* The data includes performances by a total of 10 drummers, with more than 80%
+of duration coming from hired professionals. The professionals were able to
+improvise in a wide range of styles, resulting in a diverse dataset.
+* The drummers were instructed to play a mix of long sequences (several minutes
+of continuous playing) and short beats and fills.
+* Each performance is annotated with a genre (provided by the drummer), tempo,
+and anonymized drummer ID.
+* Most of the performances are in 4/4 time, with a few examples from other time
+signatures.
+* Four drummers were asked to record the same set of 10 beats in their own
+style. These are included in the test set split, labeled eval-session/groove1-10.
+* In addition to the MIDI recordings that are the primary source of data for the
+experiments in this work, the authors captured the synthesized audio outputs of
+the drum set and aligned them to within 2ms of the corresponding MIDI files.
+
+A train/validation/test split configuration is provided for easier comparison of
+model accuracy on various tasks.
+
+The dataset is made available by Google LLC under a Creative Commons
+Attribution 4.0 International (CC BY 4.0) License.
+
+For more details, please visit: http://magenta.tensorflow.org/datasets/groove
+"""
+
+from __future__ import division
+from __future__ import print_function
+
+import csv
+import jams
+import librosa
+import logging
+import numpy as np
+import os
+import pretty_midi
+
+import mirdata.download_utils as download_utils
+import mirdata.jams_utils as jams_utils
+import mirdata.track as track
+import mirdata.utils as utils
+
+
+DATASET_DIR = 'Groove-MIDI'
+
+
+AUDIO_MIDI_REMOTE = download_utils.RemoteFileMetadata(
+    filename='groove-v1-0.0.zip',
+    url='http://storage.googleapis.com/magentadata/datasets/groove/groove-v1.0.0.zip',
+    checksum='99db7e2a087761a913b2abfb19e86181',
+    destination_dir=None,
```

Comment:
I'm having some issues with the filesystem here.
The file `groove-v1-0-0.zip` unzips to a folder named `groove`

So the dataset structure is 
`~/mir_datasets/Groove-MIDI/groove/drummer1`
`~/mir_datasets/Groove-MIDI/groove/drummer2`
`~/mir_datasets/Groove-MIDI/groove/drummer3`

etc.

Perhaps you could remove this intermediate `groove` subfolder? (ideally, without moving heavy files around ...)

Reply:
My workaround was to do `os.rename` after unzipping the file :)